### DISCUSSION TITLE
Source code for so-allow???

### DISCUSSION THREAD

**Discussion by FritzLott** [Security-Onion-Solutions/securityonion#5745] [General] [answered]
This is an awkward question because there can be no community response.

I was hoping to understand the workings of the firewall better by examining the firewall commands embedded in so-allow.  I am assuming that the so-allow binary file might be compiled python code or a compiled bash script and that as such an appropriate source file might be readily available.  

Please overlook my sloppy etiquette with this question if such a source file is not easily available.  I understand that the "glue" holding together all the Security Onion software packages is not a trivial undertaking.

**Comment by clairmont32** [comment, marked as answer]
Pretty much all of the so-* commands live here:
https://github.com/Security-Onion-Solutions/securityonion/tree/master/salt%2Fcommon%2Ftools%2Fsbin

Most of it that I've looked over is bash or python.
